src/algebra/continued_fractions/definitions/basic.lean

```diff
@@ -0,0 +1,285 @@
+
+local attribute [instance] seq.coe_to_seq
+
+/-- Coerce a gcf by elementwise coercion. -/
+instance has_coe_to_gcf : has_coe (gcf α) (gcf β) := ⟨λ ⟨h, s⟩, ⟨(h : β), (s : seq $ gcf.pair β)⟩⟩
+
+@[simp, move_cast]
+lemma coe_to_gcf {g : gcf α} : (↑(g : gcf α) : gcf β) = ⟨(g.h : β), (g.s : seq $ gcf.pair β)⟩ :=
+by { cases g, refl }
+
+end coe
+end gcf
+
+/-!
+We now continue to define simple continued fractions. A *simple continued fraction* (scf) is a gcf
+whose partial numerators are all equal to one.
+
+                                1
+                h + --------------------------
+                                  1
+                      b₀ + --------------------
+                                    1
+                            b₁ + --------------
+                                        1
+                                  b₂ + --------
+                                      b₃ + ...
+
+
+Again, for convenience, one writes `[h; b₀, b₁, b₂,...]`.
+-/
+
+/-- A gcf is a simple continued fraction (scf) if all partial numerators are equal to one. -/
+def gcf.is_scf (g : gcf α) [has_one α] : Prop :=
+∀ (n : ℕ) (aₙ : α), g.partial_numerators.nth n = some aₙ → aₙ = 1
+
+variable (α)
+/--
+A simple continued fraction (scf) is a gcf whose partial numerators are equal to one.
+It is the subtype of gcfs that satisfy `gcf.is_scf`.
+ -/
+def scf [has_one α] := {g : gcf α // g.is_scf}
+
+variable {α}
+/- Interlude: define some expected coercions. -/
+namespace scf
+variable [has_one α]
+
+/-- Lift a scf to a gcf using the inclusion map. -/
+instance has_coe_to_gcf : has_coe (scf α) (gcf α) := by {unfold scf, apply_instance}
+
+@[simp, elim_cast]
+lemma coe_to_gcf {s : scf α} : (↑s : gcf α) = s.val := rfl
+
+end scf
+
+/-!
+Next, we define regular continued fractions. A *regular continued fraction* (rcf) is a scf whose
+partial denominators `bᵢ` are all positive, i.e. `0 < bᵢ`.
+-/
+
+/-- A scf is a regular continued fraction (rcf) if all partial denominators are positive. -/
+def scf.is_rcf [has_one α] [has_zero α] [has_lt α] (s : scf α) : Prop :=
+∀ (n : ℕ) (bₙ : α), (↑s : gcf α).partial_denominators.nth n = some bₙ → 0 < bₙ
+
+variable (α)
+
+/--
+A regular continued fraction (rcf) is a scf whose partial denominators are all positive.
+It is the subtype of scfs that satisfy `scf.is_rcf`.
+ -/
+def rcf [has_one α] [has_zero α] [has_lt α] := {s : scf α // s.is_rcf}
+
+variable {α}
+
+/- Interlude: define some expected coercions. -/
+namespace rcf
+variables [has_one α] [has_zero α] [has_lt α]
+
+/-- Lift a rcf to a scf using the inclusion map. -/
+instance has_coe_to_scf : has_coe (rcf α) (scf α) := by {unfold rcf, apply_instance}
+
+@[simp, elim_cast]
+lemma coe_to_scf {r : rcf α} : (↑r : scf α) = r.val := rfl
+
+/-- Lift a rcf to a scf using the inclusion map. -/
+instance has_coe_to_gcf : has_coe (rcf α) (gcf α) := ⟨λ r, ↑(↑r : scf α)⟩
+
+@[simp, elim_cast]
+lemma coe_to_gcf {r : rcf α} : (↑r : gcf α) = r.val := rfl
+
+end rcf
+
+/-!
```

Comment:
I *think* the intention for module docs `/-! ... -/` is that there is exactly one per file, at the top of the file.